src/aliceVision/multiview/essentialF10Solver.hpp

```diff
@@ -0,0 +1,235 @@
+// This file is part of the AliceVision project.
+// Copyright (c) 2018 AliceVision contributors.
+// Copyright (c) 2015 Jan Heller <hellej1@cmp.felk.cvut.cz>, Zuzana Kukelova <zukuke@microsoft.com>
+// This Source Code Form is subject to the terms of the Mozilla Public License,
+// v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include <aliceVision/numeric/numeric.hpp>
+
+#include <Eigen/Dense>
+
+#include <cmath>
+#include <limits>
+#include <vector>
+
+namespace aliceVision {
+
+using Mat21 = Eigen::Matrix<double, 2, 1>;
+
+template <typename Scalar>
+bool isZero(Scalar val)
+{
+  return (+val < (100.0 * std::numeric_limits<Scalar>::epsilon())) &&
+         (-val < (100.0 * std::numeric_limits<Scalar>::epsilon()));
+}
+
+/**
+ * @brief	Computes column echelon form of a matrix
+ * @author Jan Heller, adapted to aliceVision by Michal Polic
+ */
+template <typename Derived>
+inline void colEchelonForm(Eigen::MatrixBase<Derived>& M, double pivtol = 1e-12)
```

Comment:
since these are help functions very specific to the algorithm, they can be just "hidden" in the cpp and not exposed in the hpp. 
Another solution is to move them in numeric as that is the module with all these kinds of functions

Reply:
if help functions are moved in the cpp, they cannot be inline anymore no ?  
Maybe they don't need to be inline....
In this case what do you think is the best idea? 
@simogasp 
@fabiencastan

Reply:
not sure I understand.
You can move these template functions at the beginning of the cpp file, wherever they will be called afterward they will be instantiated/inlined or whatever. The only drawback is that you cannot use them elsewhere or expose them as API  (they are not visible).

Reply:
ok thanks